src/Serializer/AbstractItemNormalizer.php

```diff
@@ -308,7 +308,17 @@ protected function denormalizeRelation(string $attributeName, PropertyMetadata $
             $context['resource_class'] = $className;
             $context['api_allow_update'] = true;
 
-            return $this->serializer->denormalize($value, $className, $format, $context);
+            try {
+                return $this->serializer->denormalize($value, $className, $format, $context);
+            } catch (InvalidArgumentException $e) {
+                if ('Invalid value provided (invalid IRI?).' !== $e->getMessage()) {
```

Comment:
Maybe it could be better to introduce a new Exception extending the `InvalidArgumentException`.
That said, we wouldn't to the check the message, WDYT ?
https://github.com/api-platform/core/blob/master/src/Serializer/AbstractItemNormalizer.php#L164

Reply:
Yep, added a comment on that on my description, but IMO this is not really the scope of this PR, right ? Or should I change it anyway ?